db/migrate/20210125173413_add_app_creation_workflow_to_sources.rb

```diff
@@ -0,0 +1,5 @@
+class AddAppCreationWorkflowToSources < ActiveRecord::Migration[5.2]
+  def change
+    add_column :sources, :app_creation_workflow, :string, :default => "paranoid"
```

Comment:
what about to switch to postgres enum?
i.e. https://medium.com/@diegocasmo/using-postgres-enum-type-in-rails-799db99117ff

Reply:
Initially I thought about doing that, but we did it for something similar on Catalog and it ended up causing nothing but headaches since AR doesn't play very nicely with the postgres enum type. Things get tough any time you need to change the data or add to it and it makes some AR operations not quite work (we had problems with rake db:rollback etc if I remember right) 

At the end of the day this is a pretty low-traffic table, the field only gets validated on create/update so I think keeping it a simple string would be better, the validation will make it behave as an enum anyway.

Reply:
@slemrmartin enum makes a lot of sense, except that AR and postgres enum together are a disaster. I implemented enum in Catalog ( like @lindgrenj6 references ) and we fought it the entire time after.  

I am for keeping this simpler string based validation as AR will do the right thing.